package/MDAnalysis/transformations/boxdimensions.py

```diff
@@ -92,3 +93,72 @@ def __init__(self,
     def _transform(self, ts):
         ts.dimensions = self.dimensions
         return ts
+
+
+class set_variable_dimensions(TransformationBase):
+    """
+    Set simulation box dimensions that may vary over time.
+    This is useful for creating NPT trajectories.
+
+    Timestep dimensions are modified in place.
+
+    Example
+    -------
+
+    e.g. set simulation box dimensions to a vector containing unit cell
+    dimensions [*a*, *b*, *c*, *alpha*, *beta*, *gamma*] at the first frame,
+    and [*2a*, *2b*, *2c*, *alpha*, *beta*, *gamma*] at the second frame.
+    Lengths *a*, *b*, *c* are in the MDAnalysis length unit (Å), and angles
+    are in degrees.
+
+    .. code-block:: python
+
+        dim = [
+            [2, 2, 2, 90, 90, 90],
+            [4, 4, 4, 90, 90, 90],
+        ]
+        transform = mda.transformations.boxdimensions.set_dimensions(dim)
+        u.trajectory.add_transformations(transform)
+
+    Parameters
+    ----------
+    dimensions: iterable of floats
+        vector that contains unit cell lengths and angles.
+        Expected shape is (N, 6), where N is the number of frames
+        in the trajectory.
+
+    Returns
+    -------
+    :class:`~MDAnalysis.coordinates.timestep.Timestep` object
+    """
+
+    def __init__(self,
+                 dimensions,
+                 max_threads=None,
+                 parallelizable=True):
+        super().__init__(max_threads=max_threads,
+                         parallelizable=parallelizable)
+        self.dimensions = dimensions
+
+        try:
+            self.dimensions = np.asarray(self.dimensions, np.float32)
```

Comment:
I thought box was `np.float64`?

Reply:
I copied that from [the existing `set_dimensions`](https://github.com/MDAnalysis/mdanalysis/blob/d5a30e9ad868b3b25eb27dd1df0e70cda983d8bd/package/MDAnalysis/transformations/boxdimensions.py#L77), and I just checked and the dimensions are indeed `float32`. But I think you're right - they probably should be `np.float64`?

Reply:
I think they're stored as 32s, but promoted for distance functions to 64, it's messy.

Reply:
ah okay, so is it okay to leave it as 32 here?